pkg/client/optimizely_user_context.go

```diff
@@ -0,0 +1,111 @@
+/****************************************************************************
+ * Copyright 2020, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package client //
+package client
+
+import (
+	"sync"
+
+	"github.com/optimizely/go-sdk/pkg/decide"
+	"github.com/optimizely/go-sdk/pkg/entities"
+)
+
+// OptimizelyUserContext defines user contexts that the SDK will use to make decisions for.
+type OptimizelyUserContext struct {
+	optimizely  *OptimizelyClient
+	userContext entities.UserContext
```

Comment:
Can we replace this "userContext" with userId and attributes, then we do not need to mix UserContext and OptimizelyUserContext. Keep using UserContext for legacy APIs and use OptimizelyUserContext for decide APIs.